src/pymmcore_widgets/useq_widgets/_time.py

```diff
@@ -36,10 +36,57 @@ def __init__(self, rows: int = 0, parent: QWidget | None = None):
 
         self.valueChanged.connect(self._on_value_changed)
 
-        # self.leave_shutter_open = QCheckBox("Leave shutter open across time points")
-        # self.leave_shutter_open.toggled.connect(self.valueChanged)
+    # ------------------------- Public API -------------------------
 
-        # self.layout().addWidget(self.leave_shutter_open)
+    def value(
+        self, exclude_unchecked: bool = True
+    ) -> MultiPhaseTimePlan | TIntervalLoops | TIntervalDuration:
+        """Return the current value of the table as a [useq.TimePlan](https://pymmcore-plus.github.io/useq-schema/schema/axes/#time-plans).
+
+        Returns
+        -------
+        MultiPhaseTimePlan | TIntervalLoops | TIntervalDuration
+            The current [useq.TimePlan](https://pymmcore-plus.github.io/useq-schema/schema/axes/#time-plans)
+            value of the table.
+        """
+        duration_col = self.table().indexOf(self.DURATION)
+        active_key = "duration" if self._mode_column == duration_col else "loops"
+        phases = [
+            {"interval": p["interval"], active_key: p[active_key]}
+            for p in self.table().iterRecords(exclude_unchecked=exclude_unchecked)
+        ]
+        plan = MultiPhaseTimePlan(phases=phases)
+        return plan.phases[0] if len(plan.phases) == 1 else plan  # type: ignore
+
+    def setValue(self, value: Any) -> None:
+        """Set the current value of the table from a [useq.TimePlan](https://pymmcore-plus.github.io/useq-schema/schema/axes/#time-plans).
+
+        Parameters
+        ----------
+        value : MultiPhaseTimePlan | TIntervalLoops | TDurationLoops | TIntervalDuration | None
+            The [useq.TimePlan](https://pymmcore-plus.github.io/useq-schema/schema/axes/#time-plans) to set.
+        """  # noqa: E501
```

Comment:
remove noqa and fix the problem

Reply:
this is just because of `| None` in value types. Not sure how can fix this by going on another line.